### PR TITLE
NativeCallable bugfixes

### PR DESCRIPTION
**Pull request by jkotas**
- Fix race condition in JIT_ReversePInvokeEnter
- Disable R2R for x86 on all platforms

### CODE REVIEW

**Comment by jkotas** [comment]
These are bug fixes from the large native callable change that was rolled back.

File: src/coreclr/src/vm/jithelpers.cpp [hunk 1]
```diff
@@ -5383,8 +5383,14 @@ NOINLINE static void JIT_ReversePInvokeEnterRare(ReversePInvokeFrame* frame)
     if (thread->PreemptiveGCDisabled())
         ReversePInvokeBadTransition();
 
-    thread->DisablePreemptiveGC();
     frame->currentThread = thread;
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/src/vm/jithelpers.cpp]: >frame->currentThread = thread; [](start = 4, length = 30)

Did this need to move before the call to `thread->DisablePreemptiveGC();`?

**Review by AaronRobinsonMSFT — APPROVED** [review]
:shipit:

File: src/coreclr/src/vm/jithelpers.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/src/vm/jithelpers.cpp]: `frame->currentThread` is set before calling `DisablePreemptiveGC` in the other path. I have done this to make both paths symetric.

**Pull request merged**